src/test/java/com/ci/GitUtilsTest.java

```diff
@@ -0,0 +1,161 @@
+        
+        //The redirected output from the difference between the branches has to be 
+        //greater than 0 since they contain different code.
+        File diff = new File("./target/diff.txt");
+        assertTrue(diff.length() > 0);
+    }
+
+
+    @Test
+    /**
+     * Requirements: See `updateTarget` documentation
+     * Contract:
+     *      Precondition: There is an invalid branch name passed to the `updateTarget`
+     *                    method with a valid URL and default branch name.
+     *      Postcondition: `updateTarget` throws `java.lang.Exception.class`
+     */
+    public void testUpdateTargetInvalidBranch() throws Exception{
+        //Setup arguments for updateTarget with an invalid checkout branch.
+        String testCloneURLSelf = "https://github.com/DD2480-Group31/Continuous-Integration.git";
+        String testBranchSelf = "IDoNotExist"; //Branch does not exist
+        String testDefaultBranch = "master";
+
+        //Check that the invalid call throws `java.lang.Exception.class`
+        assertThrows(java.lang.Exception.class, () -> GitUtils.updateTarget(testCloneURLSelf, testBranchSelf, testDefaultBranch));
+    }
+
+
+
+    @Test
+    /**
+     * Requirements: See `updateTarget` documentation
+     * Contract:
+     *      Precondition: There is an invalid `default` branch name passed to
+     *                    `updateTarget` with a valid URL and branch.
+     *      Postcondition: `updateTarget` throws `java.lang.Exception.class`
+     */
+    public void testUpdateTargetInvalidMain() throws Exception{
+        //Setup arguments for updateTarget with an invalid `default` branch.
+        String testCloneURLSelf = "https://github.com/DD2480-Group31/Continuous-Integration.git";
+        String testBranchSelf = "assessment";
+        String testDefaultBranch = "main"; //branch does not exist, master is the default branch.
+
+        //Check that the invalid call throws `java.lang.Exception.class`
+        assertThrows(java.lang.Exception.class, () -> GitUtils.updateTarget(testCloneURLSelf, testBranchSelf, testDefaultBranch));
+    }
+
+
+
+    @Test
+    /**
+     * Requirements: See `updateTarget` documentation
+     * Contract:
+     *      Precondition: There is an invalid URL passed to the `updateTarget` method
+     *                    with a valid branch and default branch for cloning this repository.
+     *      Postcondition: `updateTarget` throws `java.lang.Exception`
+     */
+    public void testUpdateTargetInvalidURL() throws Exception{
+        
+        //Setup arguments for updateTarget with an invalid URL.
+        String testCloneURLSelf = "invalidURL"; //Invalid
+        String testBranchSelf = "assessment";
+        String testDefaultBranch = "master";
+
+        //Invalid URL passed to updateTarget --> throw `IllegalArgumentException`
+        assertThrows(IllegalArgumentException.class, () -> GitUtils.updateTarget(testCloneURLSelf, testBranchSelf, testDefaultBranch));
```

Comment:
Since we already have the target directory, the function will not perform a clone. You can probably use the `cloneRepo` function instead for the test.

Reply:
Yes but I have included dependencies to check the URL and it now throws an IllegalArgumentException when the URL is invalid. Should I still change it?